LibJpegWrapper/TJDecompressor.cs

```diff
@@ -44,54 +59,80 @@ public TJDecompressor()
         /// <returns>Raw pixel data of specified format</returns>
         /// <exception cref="TJException">Throws if underlying decompress function failed</exception>
         /// <exception cref="ObjectDisposedException">Object is disposed and can not be used anymore</exception>
-        public unsafe byte[] Decompress(IntPtr jpegBuf, ulong jpegBufSize, TJPixelFormats destPixelFormat, TJFlags flags, out int width, out int height, out int stride)
+        public byte[] Decompress(IntPtr jpegBuf, ulong jpegBufSize, TJPixelFormats destPixelFormat, TJFlags flags, out int width, out int height, out int stride)
         {
-            int outBufSize;
-            this.GetImageInfo(jpegBuf, jpegBufSize, destPixelFormat, out width, out height, out stride, out outBufSize);
+            if (_isDisposed)
+                throw new ObjectDisposedException("this");
 
-            var buf = new byte[outBufSize];
+            int outBufSize;
+            GetImageInfo(jpegBuf, jpegBufSize, destPixelFormat, out width, out height, out stride, out outBufSize);
 
-            fixed (byte* bufPtr = buf)

```

Comment:
I believe this change introduces a performance regression.

In the original code, one buffer (`buf`) was allocated in managed code which was pinned and shared with unmanaged code.

In the changed code, you first create a new buffer in unmanaged code (`new TJUnmanagedMemory(bufferSize)`).
When you return `outBuf`, you implicitly cast it to a `byte[]`. That operator allocates a new byte array and returns that one.

Net, in the original code you had one allocation of a large object, in the changed code you have two allocations (one in unmanaged memory and one in managed memory); plus a copy operation.

This code is on a hot path for us so performance is important.

Perhaps using `Span<T>` may be a solution here, not sure.
If you think there's no performance difference, feel free to send over a BenchmarkDotNet test which compares both ;-).